compiler/rustc_resolve/src/ident.rs

```diff
@@ -793,6 +799,7 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         // This binding should be ignored during in-module resolution, so that we don't get
         // "self-confirming" import resolutions during import validation and checking.
         ignore_binding: Option<NameBinding<'a>>,
+        used: Option<Used>,
```

Comment:
This parameter only makes sense when the `finalize: Option<Finalize>` parameter above is `Some`.
So instead of adding this parameter, `used: Used` can be added to `struct Finalize` and initialized to `Used::Other` in `Finalize::with_root_span`.

This could be less convenient than the current setup though.

Reply:
If it's not moved to `Finalize`, then `resolve_ident_in_module_unadjusted_ext` is the only function that needs this parameter.
`resolve_ident_in_module_unadjusted` doesn't need it, it's always `Used::Other` there.
Could you also move the `used` parameter between `restricted_shadowing` and `finalize` in that case.

Reply:
Ok, moving it into `Finalize` looks good.